### DISCUSSION TITLE
includeList default order?

### DISCUSSION THREAD

**Discussion by jason-downs** [serverpod/serverpod#2388] [Q&A] [answered]
Querying a 1:n relation:

 In the absense of specifying an `orderBy` argument in `includeList()` is it supposed to guarantee a default order, such as by ID, or is it just random? I had been expecting it to return the list sorted by ID by default, which is the order that the items were originally written to that list, but I was occasionally getting unexpected results where the order seemed random. It is simple for me to fix it by specifiying `orderBy` to sort by the ID, but I was just wondering if this was the desired behavior or if no specific ordering is guaranteed without passing in an argument for `orderBy` .

It seems the occasional unexpected ordering (where it's not ordered by ID by default) started recently but it may just be that I only recently noticed.

**Comment by SandPod** [comment, marked as answer]
Hey @jason-downs!

If `orderBy` is not specified the rows are returned in an unspecified order as described in the [Postgres documentation](https://www.postgresql.org/docs/current/queries-order.html#:~:text=ASC%20order%20is%20the%20default,is%20determined%20with%20the%20%3E%20operator.).

More specifically this is how the ordering described in the documentation:

> If sorting is not chosen, the rows will be returned in an unspecified order. The actual order in that case will depend on the scan and join plan types and the order on disk, but it must not be relied on. A particular output ordering can only be guaranteed if the sort step is explicitly chosen.
